app/src/main/java/com/concordium/wallet/data/walletconnect/Schema.kt

```diff
@@ -1,6 +1,20 @@
 package com.concordium.wallet.data.walletconnect
 
-data class Schema(
-    val type: String?,
-    val value: String?
-)
+sealed class Schema {
+    data class ValueSchema(
+        val type: String?,
+        val value: String?
+    ) : Schema()
+
+    /**
+     * For unknown reasons sometimes the schema comes in bad format, after some research it looks
+     * like a broken buffer issue, in order to not crash and successfully handle transactions
+     * we handle this specific case
+     */
```

Comment:
```suggestion
In some cases the buffer object in Java Script has been serialized directly, instead of first converting the buffer object to a hex string and thereafter converting the string to a byte array.
```